core/base/topoMap/TopoMap.h

```diff
@@ -0,0 +1,1045 @@
+/// Visualization and Computer Graphics 27(2): 561-571, 2020.
+
+/// \sa TopoMap
+
+#pragma once
+
+// ttk common includes
+#include <Debug.h>
+#include <Geometry.h>
+#include <UnionFind.h>
+
+// STL includes
+#include <map>
+#include <queue>
+#include <set>
+#include <sstream>
+#include <utility> // For std::pair
+#include <vector>
+
+// cos and sin computations, so we are cautious on testing equalities between
+// double.
+static double Epsilon{ttk::Geometry::pow(10.0, -FLT_DIG + 2)};
+static double EpsilonDBL{ttk::Geometry::pow(10.0, -DBL_DIG + 2)};
+
+template <typename T>
+static inline T compute_dist2(const T *ptA, const T *ptB) {
+  T dx = ptB[0] - ptA[0], dy = ptB[1] - ptA[1];
+
+  return dx * dx + dy * dy;
+}
+
+template <typename T>
+static inline T compute_dist(const T *ptA, const T *ptB) {
+  T dx = ptB[0] - ptA[0], dy = ptB[1] - ptA[1];
+
+  return sqrt(dx * dx + dy * dy);
+}
+
+template <typename T>
+inline bool are_colinear(const T *pptA, const T *pptB, const T *pptC) {
+  double ptA[2] = {pptA[0], pptA[1]}, ptB[2] = {pptB[0], pptB[1]},
+         ptC[2] = {pptC[0], pptC[1]};
+  return fabs(ptA[0] * (ptB[1] - ptC[1]) + ptB[0] * (ptC[1] - ptA[1])
+              + ptC[0] * (ptA[1] - ptB[1]))
+         <= EpsilonDBL;
+}
+
+// Normalizes a given vector.
+template <typename T>
+static void
+  computeUnitVector(const T *coordOrig, const T *coordDest, T *coordVect);
+
+// Computes the oriented angle BAC
+template <typename T>
+static double computeAngle(const T *ptA, const T *ptB, const T *ptC);
+
+// Rotates the first argument by angle around center.
+template <typename T>
+static void rotate(T *ptToRotate, const T *center, double angle);
+
+// Rotate the set of points by the given angle, considering the given center as
+// center of the rotation.
+template <typename T>
+static void
+  rotatePolygon(std::vector<T> &coords, T *centerCoords, const double angle);
+
+// Real call to the convex hull engine: either Qhull or Boost.
+bool computeConvexHull_aux(const std::vector<double> &coords,
+                           std::vector<size_t> &res,
+                           std::string &errMsg);
+
+namespace ttk {
+
+  /**
+   * The TopoMap class computes a 2D projection of the points given as input,
+   * either by their distance matrix or by their coordinates.
+   */
+  class TopoMap : virtual public Debug {
+
+  public:
+    /** We rely on constructing a minimal spanning tree. We allow for the two
+     * main algorithms to build one: Kruskal and Prim.
+     */
+    enum class STRATEGY { KRUSKAL = 0, PRIM = 1 };
+
+    TopoMap();
+    TopoMap(size_t angularSampleNb, bool checkMST, STRATEGY strategy)
+      : AngularSampleNb(angularSampleNb), CheckMST(checkMST),
+        Strategy(strategy) {
+
+      // inherited from Debug: prefix will be printed at the beginning of every
+      // msg
+      this->setDebugMsgPrefix("TopoMap");
+    }
+    ~TopoMap() override;
+
+    /**
+     * @brief Computes the topological mapper
```

Comment:
thanks for rephrasing all occurrences of the expression "topological mapper".
here, I'd suggest to rephrase into "Computes the TopoMap projection".